app/controllers/sp_components_controller.rb

```diff
@@ -13,6 +13,7 @@ def new
 
   def show
     @component = SpComponent.find(params[:id])
+    @available_services = Service.all.select { |s| s.sp_component_id.blank? }
```

Comment:
@available_services = Service.all.select { |s| s.sp_component_id.blank? }

I don't think a select is good here because it going to first get all the services in memory then pass the block to do the filtering. I think a where would be more appropriate.
```
 @available_services = Servicewhere(sp_component_id: [nil, '']
```
Something like the above